src/validators/date.rs

```diff
@@ -153,7 +153,7 @@ fn date_from_datetime<'data>(
         minute: 0,
         second: 0,
         microsecond: 0,
-        tz_offset: None,
+        tz_offset: dt.time.tz_offset,
```

Comment:
@samuelcolvin what are your thoughts on this? It means that we allow `datetime(2023, 7, 13, 0, 0, 0, tzinfo=timezone.utc)` -> `date(2023, 7, 13)`. We need it because when we parse a unix timestamp to a date it goes through a datetime first. I can try to work around it but maybe it's reasonable as is.

Reply:
I think it seems fine.

Simpler might be to just do `dt.time.hour == 0 && ... && dt.time.microsecond == 0`?